builder/evaluator.go

```diff
@@ -191,6 +197,18 @@ func (b *builder) Run(context io.Reader) (string, error) {
 		}
 	}
 
+	// check if there are any leftover build-args that were passed but not
+	// consumed during build. Return an error, if there are any.
+	leftoverArgs := []string{}
```

Comment:
Actually thinking more there is a gotcha with doing this check before dispatch in the case when `b.Config.AllowedBuildArgs` are going to be inherited from parent image. In that case just looking at pre-parsed dockerfile will error incorrectly. I think doing this check post build is correct.